uplink/build.rs

```diff
@@ -0,0 +1,46 @@
+use std::{
+    fs::File,
+    io::{Read, Write},
+    process::Command,
+};
+use which::which;
+
+#[cfg(not(windows))]
+fn main() {
+    if let Ok("debug") = std::env::var("PROFILE").as_deref() {
+        //~/.rover/bin/rover graph introspect https://uplink.api.apollographql.com/
+        which("rover")
+    .map_err(|_| "could not find path to rover executable, see installation instructions at https://github.com/apollographql/rover#installation-methods").unwrap();
+
+        let output = Command::new("rover")
+            .args([
+                "graph",
+                "introspect",
+                "https://uplink.api.apollographql.com/",
+            ])
+            .output()
+            .expect("failed to execute process");
+
+        let mut buf = Vec::new();
+        let _ = File::open("uplink.graphql")
+            .unwrap()
+            .read_to_end(&mut buf)
+            .unwrap();
+        let data = output.stdout;
+        assert_eq!(
+            std::str::from_utf8(&buf).unwrap(),
+            std::str::from_utf8(&data).unwrap(),
+            "Uplink schema changed"
+        );
+
+        File::create("uplink.graphql")
```

Comment:
Logic here opens the file, checks if it identical and then writes it out again even though it's identical. I guess just remove the code to write out the schema.